Automattic/vip-actions
# Port dependabot PR automerge from `vip-go-crontab` to Github Actions

## Description

As per title, this PR ports the code from `vip-go-crontab` to something that GitHub Actions would like.

We use the hydrated Octokit from GitHub instead of raw REST APIs, and we use auto-merge instead so that we don't need additional permission for getting the list of checks that needs to pass before merging.

## References:

[Previous PR from vip-go-crontab](https://github.com/Automattic/vip-go-crontab/pull/1149)

## How to test:

1. Setup a fresh `npm` project repo i.e. by using `npm init`
2. Commit code to github into your personal repo
4. Follow the `README.md` [in this](https://github.com/Automattic/vip-actions/blob/042878ab81e68f31d20076a82aeec44314c94069/dependabot-auto-merge/README.md) PR.
5. Once followed, you could head on to your repo, press the 'Actions' tab, click Dependabot auto-merge, press Run Workflow on the default branch.
6. Test away! It should be able to merge dependabot PRs that consists of only minor version change(s), and only those that has all tests passing. It should not merge PRs that has zero passing checks, even if there's no workflows created yet.
